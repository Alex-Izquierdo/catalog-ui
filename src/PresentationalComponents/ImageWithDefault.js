import React from 'react';
import propTypes from 'prop-types';

const ImageWithDefault = ({src, defaultSrc, ...other}) => {
  let element;

  const changeSrc = newSrc => {
    element.src = newSrc;
  };

  return (
      <React.Fragment>
        <img src={src}
             onError={() => changeSrc(defaultSrc)}
             ref={el => element=el}
             {...other} />
      </React.Fragment>
  );
};

ImageWithDefault.propTypes = {
  src: propTypes.string,
  defaultSrc: propTypes.string
};

export default ImageWithDefault;
