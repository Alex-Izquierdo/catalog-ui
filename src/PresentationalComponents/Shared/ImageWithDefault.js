import React from 'react';
import propTypes from 'prop-types';
import DefaultPortfolioImg from '../../assets/images/default-portfolio.jpg';

const ImageWithDefault = ({ src, ...props }) => <img src={ src } { ...props } />;

ImageWithDefault.propTypes = {
  src: propTypes.string
};

ImageWithDefault.defaultProps = {
  src: DefaultPortfolioImg
};

export default ImageWithDefault;
