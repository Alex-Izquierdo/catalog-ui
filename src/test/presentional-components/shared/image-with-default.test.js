import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import ImageWithDefault from '../../../PresentationalComponents/Shared/ImageWithDefault';

describe('<ImageWithDefault />', () => {
  let initialProps;
  beforeEach(() => {
    initialProps = {
      src: 'foo'
    };
  });

  it('should render correctly', () => {
    const wrapper = mount(<ImageWithDefault { ...initialProps } />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
