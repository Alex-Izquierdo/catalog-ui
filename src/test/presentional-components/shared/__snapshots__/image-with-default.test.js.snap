// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageWithDefault /> should render correctly 1`] = `
<ImageWithDefault
  src="foo"
>
  <img
    src="foo"
  />
</ImageWithDefault>
`;
